src/options.rs

```diff
@@ -108,3 +101,21 @@ impl TryFrom<&Args> for Options {
         })
     }
 }
+
+fn build_glob_set(glob_set: &Vec<String>) -> Result<Option<GlobSet>>{
```

Comment:
I think this would be a reasonable thing to add unit tests for, in a `mod test` in this file: e.g. that files with and without slashes do and don't match in various combinations.

Reply:
Generally, I try to avoid writing tests coupled to the code structure and internal functions. The reason is that such things can result in fragile tests, so they can easily break in the future, in case of refactoring of the code.

What do you think about covering these cases in `cli.rs`? These are sort of edge cases, so I believe having public API tests for these can be reasonable.

Reply:
Yep, I generally feel the same. 

In this case the public interface tests are a _little_ slow because they spawn subprocesses which can take a fraction of a second. If they just do `--list` that's not so bad, and that's probably enough to test this well.